recursion/src/test/java/FactorialRecursiveImplementationTest.java

```diff
@@ -20,10 +22,18 @@
         );
     }
 
+    Factorial sut;
+
+    @BeforeEach
+    void init() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
+                       InvocationTargetException, InstantiationException {
+        sut = (Factorial) Class.forName("FactorialImpl").getDeclaredConstructor().newInstance();
```

Comment:
As I mentioned before I want to make tests completely runnable and unmodifiable before an implementation. Let's look at this as a plugin system

Reply:
Ok, unmodifiable sounds good. 

Since implementation class name in this case should be introduced in tests before implementation and it should not be modified, let's than throw away `Factorial` interface and introduce full-fledged `FactorialImpl` class with methods throwing `java.lang.UnsupportedOperationException`. As a result:
 - test classes could stay unchanged
 - no reflection in `init` method with no `ReflectiveOperationException` mentioning
 - no default method in interfaces to be implemented
 - runnable tests with runtime exceptions in place

Reply:
Fixed 9f35087